R/filter_panel_api.R

```diff
@@ -83,8 +91,26 @@ FilterPanelAPI <- R6::R6Class( # nolint
     #'
     #' @return `NULL`
     remove_all_filter_states = function(datanames) {
-      datanames_to_remove <- if (missing(datanames)) private$filtered_data$datanames() else datanames
-      private$filtered_data$remove_all_filter_states(datanames = datanames_to_remove)
+      if (private$filtered_data$get_filter_turn()) {
+        datanames_to_remove <- if (missing(datanames)) private$filtered_data$datanames() else datanames
+        private$filtered_data$remove_all_filter_states(datanames = datanames_to_remove)
+      } else {
+        warning("Filter Panel is turn off so the action can not be applied with api.")
+      }
+      invisible(NULL)
+    },
+    #' @description Toggle the state of the Filter Panel turn on/off button.
+    #' @param id (`character(1)`)\cr
+    #'  `id` to the toggle button.
+    #'
+    #' @return `NULL`
+    filter_panel_toggle = function() {
```

Comment:
It could be hard to track down by the developer. Please explain the technical process of the toggle
That a toggle of the state indirectly changes the filter states by switching button. Switch button is observed by the server which disables/enables all filter states. Disabled Filter states are saved so they are restored when filter is again enabled. One can't neither change nor add the filter states if they are disabled

Reply:
I am still a bit confused about this part mainly because I don't see the `filter_panel_toggle()` method is being called anywhere. Also, the roxygen note saying that it takes `id` as parameter but I don't see any argument.

I removed this method, `load_all()`, run teal.gallery exploratory app, and was still able to toggle the filter panel successfully. Do we actually need this method?

Reply:
You are right there is no need for id argument here, +10 points for finding. 
So the whole filter_panel_api seems to be a weight without a proper value added, especially as a exported to end users item.
If we remove toggle method method then we could simplify the FilteredData class as the ui id does not have to be recorded. In my opinion till there is no decision to stop export/support filter_panel_api till then we should not limit its usage/functionality.

Reply:
@pawelru we had some discussion about filter panel api and how far we want to support/maintain it. 
It is needed e.g. for teal.reporter and getting filter states in the modules but should it be exported and continuously improved.

Reply:
> It is needed e.g. for teal.reporter and getting filter states in the modules

For now it is :)